### ISSUE TITLE
How to use a fix test set 

### ISSUE THREAD

**Issue by abdollahpouri** [guoguibing/librec#246]
I know this issue existed before too but I guess it is still not clear how to use a fix test set. Imagine you have a trainset.txt and testset.txt and they are both in the folder called data. I want to build the model using the trainset.txt and then generate recommendations for the users in the testset.txt.  This what I have done but it does not work. The problem is that it still tries to split the data into train and test but we already have the train and test set. In the Librec console log we can see :"Split data to train Set and test Set successfully!" 

```
data.input.path=data/trainset.txt
data.testset.path=data/testset.txt
data.model.splitter=testset
```

Please clarify how this exactly could be done because in many experiments we usually use owr own testset.

**Comment by SunYatong** [comment]
"data.input.path" denotes the file or directory that includes the full data (trainSet+testSet)
So your configuration should be:
```
data.input.path=data/
data.testset.path=data/testset.txt
data.model.splitter=testset
```

**Comment by abdollahpouri** [comment]
I did that but I think both train and test set files are being used for the training in Librec even I do that. I am saying that because using the movielens dataset, after I split it to train and test files myself, and did exactly what you said in your previous response, I got a precision around 0.85  using user-KNN algorithm which is way too high for this dataset. It usually should be around 0.25.

**Comment by SunYatong** [comment]
Can you debug the algorithm and check the size of the trainMatrix to see whether it matches the size of your trainSet ?

**Comment by abdollahpouri** [comment]
As you can see from this picture, both trainMatrix and testMatrix objects have 6040* 3043  dimension.

<img width="1226" alt="screen shot 2018-03-25 at 3 11 32 pm" src="https://user-images.githubusercontent.com/19310528/37879445-e4e429dc-303e-11e8-851f-7b8145b61dd3.png">

**Comment by SunYatong** [comment]
They have the same shape because they share the same user/item mapping.
You should set the breakpoint in UserKNN, and check the number of non-zero entries of trainMatrix and testMatrix.
![image](https://user-images.githubusercontent.com/17143022/37882127-022c18bc-30d3-11e8-9d4b-f62b058afe96.png)

**Comment by abdollahpouri** [comment]
The sizes of the train and test dataset seem correct but I am sure somewhere in the code (which I don't know where) something is going wrong. For example, I tested using u1.base and u1.test in the movielens-100K dataset so you can reproduce the problem yourself and I got the following result which is obviously not correct.

<img width="760" alt="screen shot 2018-03-26 at 2 20 45 pm" src="https://user-images.githubusercontent.com/19310528/37927932-ebcb3c1e-3100-11e8-897a-5a5e9e981587.png">


You can find the dataset here: 
https://grouplens.org/datasets/movielens/100k/
once you downloaded it, use u1.base as training set and u1.test as test set and run user KNN recommender on it.

**Comment by SunYatong** [comment]
Here is my result, which seems reasonable :

![image](https://user-images.githubusercontent.com/17143022/37970535-4b00c4aa-3206-11e8-8ac8-234330beb068.png)

Here is my configuration:

```
data.model.splitter=testset
data.input.path=movielens/ml-100k
data.testset.path=movielens/ml-100k/u1.test
rec.recommender.class=itemknn
rec.similarity.class=pcc
rec.neighbors.knn.number=50
rec.recommender.similarities=item
rec.recommender.isranking=true
rec.recommender.ranking.topn=10
rec.similarity.shrinkage=10
```
And I run it with the testcase in UserKNNTestCase:

![image](https://user-images.githubusercontent.com/17143022/37970669-a52d2a54-3206-11e8-8323-5495cf11798d.png)

**Comment by abdollahpouri** [comment]
I cloned the librec again and and used your configuration and I got the same result as yours which is great. I think the Librec I had was old and there was probably something wrong going on. Thank for your help.
